fields.yml

```diff
@@ -76,6 +76,41 @@
             This id normally changes across restarts, but `agent.id` does not.
           example: 8a4f500f
     
+    - name: autonomous_system
+      title: Autonomous System
+      group: 2
+      short: Fields describing an Autonomous System (Internet routing prefix).
+      description: >
+        An autonomous system (AS) is a collection of connected Internet Protocol
+        (IP) routing prefixes under the control of one or more network operators on
+        behalf of a single administrative entity or domain that presents a common,
+        clearly defined routing policy to the internet.
+    
+      reusable:
+        top_level: false
+        expected:
+          - client
+          - destination
+          - server
+          - source
+      type: group
+      fields:
+    
+        - name: number
+          level: extended
+          type: long
+          description: >
+            Unique number allocated to the autonomous system. The autonomous
+            system number (ASN) uniquely identifies each network on the Internet.
+          example: 15169
+    
+        - name: organization.name
```

Comment:
I think `organization_name` would be a better name, unless there will be more properties than just `name` under the `organization` property.

Reply:
Ah, I missed the suggestion of `organization.country_code`. Maybe that information is better suited for the `geoip` field group, though.

Reply:
I was thinking that the `source.geo` field might be populated with the location of the IP whereas the details under `source.autonomous_system.organization` would be about the organization (like perhaps the country where the organization is headquartered.)

Reply:
We could have something like `source.autonomous_system.organization.geo`, but maybe that's a bit over the top.

Reply:
`geo` is a “reusable object” so it could be used under `organization`. The current structure in this PR won’t preclude this from specified in the schema at a later time. This PR will get us the bare minimum fields for using the geoip processor with the AS database.